pkg/webui/console/containers/notifications/notification-list/index.js

```diff
@@ -64,9 +64,19 @@ const NotificationList = ({
   // Otherwise, if totalCount is 0, it means the list is empty and we should not have a total count.
   const itemCount = totalCount >= 0 ? totalCount : 100
 
+  const handleClick = useCallback(
+    async (_, id) => {
+      const clickedNotification = items.find(notification => notification.id === id)
+      if (!isArchive && !('status' in clickedNotification) && totalUnseenCount > 0) {
+        await dispatch(attachPromise(updateNotificationStatus([id], 'NOTIFICATION_STATUS_SEEN')))
+      }
+    },
+    [items, dispatch, isArchive, totalUnseenCount],
+  )
+
```

Comment:
Instead of using the click handler, you can use a `useEffect` in `containers/notifications/index.js` that watches the `selectedNotification` and sends a "seen" request if required. Let's actually put this behind a `setTimeout` so it only updates after the notification has been "seen" for two seconds.